### PR TITLE
OSX lang node var

### PR DESCRIPTION
**Pull request by odra**
# Changes

* adds "configure_base_system" provision osx playbook;
* adds "LANG" to jenkins node template

# Verification

## Base system playbook

Run the sample playbook with the tag "osx_base_system":

`ansible-playbook -i osx-sample osx.yml --tags=osx_base_system`

It should create .profile and .bash_profile files and should add both PATH and LANG env vars into "bash_profile" file.

## Jenkins osx node

Run the sample playbook with the tag "osx_configure_buildfarm":

`ansible-playbook -i osx-sample osx.yml --tags=osx_configure_buildfarm`

It should add the already defined LANG env var into the jenkins osx node configuration.

### CODE REVIEW

**Comment by wei-lee** [comment]
@odra why do we need this change?

**Comment by odra** [comment]
We need to set the LANG env var in jenkins osx node but it was also being used by other playbooks so I moved into a playbook/var of its own.

When trying it with a new user I did notice that there was no bash_profile  or profile files so I added those as well.

**Comment by wei-lee** [comment]
👀

**Comment by wei-lee** [comment]
let's squash some of the commits and merge this.

File: provision-osx/tasks/configure_base_system.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  name: Configure bash profile language encoding
+  lineinfile:
+    dest: ~/.bash_profile
+    line: "export LANG={{ buildfarm_lang_env_var }}"
+
+-
+  name: Configure profile file
+  lineinfile:
+    dest: ~/.profile
+    line: "source ~/.bash_profile"
```
Comment (wei-lee) [inline provision-osx/tasks/configure_base_system.yml]: @odra looks like this line is causing a problem on the Mac server.  There is already a `.bash_profile` on the Mac server and it has this line:

```
[[ -s "$HOME/.profile" ]] && source "$HOME/.profile" # Load the default .profile
```

But here the `.profile` is referencing the `.bash_profile` again. It is causing circular dependency between the 2 files and the ssh login doesn't work anymore.

Is this line required? If not, might be better remove it? I think by default, all `.profile`, `.bash_profile` and `.bashrc` are all loaded anyway?

Comment (odra) [inline provision-osx/tasks/configure_base_system.yml]: I created a fresh new user in a mac machine and those bash files were not there (bashrc, bash_profile and profile), which is why I created this task - but yes, I think we can remove it for now and test it later with a fresh new user to check if those filer were created (or not).

Comment (wei-lee) [inline provision-osx/tasks/configure_base_system.yml]: @odra ok, can you create a PR to remove it? If we are going to include this, we need to make sure it won't cause circular dependencies between those files.

**Pull request merged**